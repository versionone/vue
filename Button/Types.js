'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var standard = exports.standard = 'standard';
var basic = exports.basic = 'basic';
var important = exports.important = 'important';
var alt = exports.alt = 'alt';
var basicAlt = exports.basicAlt = 'basic-alt';
var special = exports.special = 'special';