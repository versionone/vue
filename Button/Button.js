'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _colorFunctions = require('@andrew-codes/color-functions');

var _Radium = require('./../utilities/Radium');

var _Radium2 = _interopRequireDefault(_Radium);

var _Transparent = require('./../utilities/Transparent');

var _Transparent2 = _interopRequireDefault(_Transparent);

var _Transitions = require('../utilities/Transitions');

var _colorManipulator = require('./../utilities/colorManipulator');

var _component = require('./../utilities/component');

var _Sizes = require('./Sizes');

var ButtonSizes = _interopRequireWildcard(_Sizes);

var _Types = require('./Types');

var ButtonTypes = _interopRequireWildcard(_Types);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var darkenInvert = function darkenInvert(foreground, background) {
    var inverseBasicColorMultiplier = 0.35;
    var darkenedColor = (0, _colorFunctions.toRgbaString)((0, _colorFunctions.darken)(background, inverseBasicColorMultiplier));
    return {
        ':hover': {
            background: darkenedColor,
            border: '1px solid ' + darkenedColor,
            color: foreground
        },
        background: background,
        border: '1px solid ' + background,
        color: foreground
    };
};
var getStylesBasedOnType = function getStylesBasedOnType(props, theme) {
    var altColor = theme.altColor,
        basicColor = theme.basicColor,
        importantColor = theme.importantColor,
        darkInverseColor = theme.darkInverseColor,
        lightInverseColor = theme.lightInverseColor,
        normalBackground = theme.normalBackground,
        textPrimaryColor = theme.textPrimaryColor;
    var disable = props.disable,
        type = props.type;

    var inverseColors = [darkInverseColor, lightInverseColor];

    if (disable) {
        var disabledColorOpacity = 0.3;
        var color = (0, _colorFunctions.toRgbaString)((0, _colorFunctions.setOpacity)(textPrimaryColor, disabledColorOpacity));
        return {
            ':hover': {
                background: normalBackground,
                color: color
            },
            background: normalBackground,
            border: '1px solid ' + _Transparent2.default,
            color: color
        };
    }
    if (type === ButtonTypes.basic) {
        return darkenInvert(darkInverseColor, basicColor);
    }
    if (type === ButtonTypes.important) {
        return darkenInvert(darkInverseColor, importantColor);
    }
    if (type === ButtonTypes.alt) {
        return darkenInvert(darkInverseColor, altColor);
    }
    if (type === ButtonTypes.basicAlt) {
        return {
            ':hover': {
                background: basicColor,
                border: '1px solid ' + basicColor,
                color: darkInverseColor
            },
            background: normalBackground,
            border: '1px solid ' + textPrimaryColor,
            color: textPrimaryColor
        };
    }

    if (type === ButtonTypes.special) {
        return {
            ':hover': {
                background: basicColor,
                border: '1px solid ' + basicColor,
                color: darkInverseColor
            },
            background: textPrimaryColor,
            border: '1px solid ' + textPrimaryColor,
            color: normalBackground
        };
    }

    var inverseBackground = (0, _colorManipulator.getForegroundForBackground)(normalBackground, inverseColors);
    var inverseForeground = (0, _colorManipulator.getForegroundForBackground)(inverseBackground, inverseColors);
    return {
        ':hover': {
            background: inverseBackground,
            color: inverseForeground
        },
        background: normalBackground,
        border: '1px solid ' + textPrimaryColor,
        color: textPrimaryColor
    };
};
var getStyles = function getStyles(props, theme) {
    var disable = props.disable,
        size = props.size;
    var normalRadius = theme.normalRadius,
        basicFontFamily = theme.basicFontFamily,
        bold = theme.bold,
        largeLineHeight = theme.largeLineHeight,
        smallFontSize = theme.smallFontSize;


    var fontSize = smallFontSize * size;
    var height = Math.ceil(fontSize * largeLineHeight);
    var borderRadiusMultiplier = 2;
    var borderRadius = normalRadius * borderRadiusMultiplier;
    var typeStyles = getStylesBasedOnType(props, theme);

    return {
        root: _extends({
            ':focus': {
                outline: 'none'
            },
            alignItems: 'flex-start',
            border: '1px solid ' + _Transparent2.default,
            borderRadius: borderRadius + 'px',
            boxSizing: 'border-box',
            cursor: disable ? 'not-allowed' : 'pointer',
            display: 'inline-block',
            fontFamily: basicFontFamily,
            fontSize: fontSize + 'px',
            fontWeight: bold,
            height: height + 'px',
            letterSpacing: '0.03em',
            lineHeight: '' + largeLineHeight,
            margin: '0',
            padding: '0 1em',
            textAlign: 'center',
            textShadow: 'none',
            transition: (0, _Transitions.create)('0.5s'),
            whiteSpace: 'no-wrap'
        }, typeStyles)
    };
};

var Button = function Button(props, context) {
    var disable = props.disable,
        onClick = props.onClick,
        text = props.text;

    var handleClick = (0, _component.createConditionalEventHandler)(!disable)(onClick);
    var styles = getStyles(props, context.theme);
    return _react2.default.createElement(
        'button',
        {
            style: styles.root,
            onClick: handleClick
        },
        text
    );
};
process.env.NODE_ENV !== "production" ? Button.propTypes = {
    /**
     * Disables the button from responding to event handlers
     */
    disable: _react.PropTypes.bool,
    /**
     * Click event handler; fired once a button is clicked
     */
    onClick: _react.PropTypes.func,
    /**
     * Numeric value used as a multiplier to the button's size; 0.75, 1, and 1.5 as examples
     */
    size: _react.PropTypes.number,
    /**
     * Text string displayed within Button
     */
    text: _react.PropTypes.string,
    /**
     * Type of button
     */
    type: _react.PropTypes.oneOf([ButtonTypes.standard, ButtonTypes.basic, ButtonTypes.important, ButtonTypes.alt, ButtonTypes.basicAlt, ButtonTypes.special])
} : void 0;
Button.defaultProps = {
    disable: false,
    onClick: function onClick() {},
    size: ButtonSizes.normal,
    text: '',
    type: ButtonTypes.standard
};
Button.contextTypes = {
    theme: _react.PropTypes.object.isRequired
};
Button.displayName = 'Button';

exports.default = (0, _Radium2.default)(Button);