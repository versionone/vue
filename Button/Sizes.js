"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var xxSmall = exports.xxSmall = 0.375;
var xSmall = exports.xSmall = 0.5;
var small = exports.small = 0.75;
var normal = exports.normal = 1;
var large = exports.large = 1.25;