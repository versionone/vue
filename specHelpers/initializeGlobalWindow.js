import {jsdom} from 'jsdom';

export default () => {
    const exposedProperties = ['window', 'navigator', 'document'];
    global.document = jsdom('<!doctype html><html><body></body></html>');
    global.window = document.defaultView;
    global.navigator = {
        userAgent: 'node.js'
    };
    Object.keys(document.defaultView).forEach((property) => {
        if (typeof global[property] === 'undefined') {
            exposedProperties.push(property);
            global[property] = document.defaultView[property];
        }
    });
}
