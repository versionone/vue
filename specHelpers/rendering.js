import toJson from 'enzyme-to-json';
import React, {PropTypes} from 'react';
import {combineReducers, createStore} from 'redux';
import Enzyme, { mount, render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import {reducer as uiReducer} from 'redux-ui';
import theme from './../src/styles/themes/v1Theme';

Enzyme.configure({ adapter: new Adapter() });

const reducer = combineReducers({
    ui: uiReducer,
});
const store = createStore(reducer);
const ctx = {
    store,
    theme,
    runQuery: jest.fn().mockReturnValue(Promise.resolve([])),
};

export const getShallow = Component => (props = {}, context = {}) => shallow(<Component {...props} />, {
    context: {
        ...ctx,
        ...context,
    },
    childContextTypes: {
        store: PropTypes.object,
        theme: PropTypes.object,
        runQuery: PropTypes.func,
    },
});

export const getRender = Component => (props = {}, context = {}) => render(<Component {...props} />, {
    context: {
        ...ctx,
        ...context,
    },
    childContextTypes: {
        store: PropTypes.object,
        theme: PropTypes.object,
        runQuery: PropTypes.func,
    },
});

export const getMount = Component => (props = {}, options = {}, context = {}) => mount(<Component {...props} />, {
    context: {
        ...ctx,
        ...context,
    },
    childContextTypes: {
        store: PropTypes.object,
        theme: PropTypes.object,
        runQuery: PropTypes.func,
    },
    ...options,
});

const replaceUIKey = componentName => {
    let index = 0;
    return json => {
        if (!Boolean(json.props) || !Boolean(json.props.uiKey)) {
            return json;
        }
        const uiKey = `${componentName}-${index}`;
        json.props.uiKey = uiKey;
        json.props.uiPath = [uiKey];
        index += 1;
        return json;
    };
};
const replaceUiKeyRecursively = (json) => {
    if (!Boolean(json.type)) {
        return json;
    }
    const replaceUIKeyInParent = replaceUIKey(json.type);
    json = replaceUIKeyInParent(json);
    if (!Boolean(json.children)) {
        return json;
    }
    json.children = json.children.map((child, index) => replaceUiKeyRecursively(child));
    return json;
};

const removeReduxUIGeneratedParts = (json) => {
    if (!Boolean(json.children)
        || json.children === 0
        || json.children[0].type !== 'UI'
        || !Boolean(json.children[0].props)
        || !Boolean(json.children[0].props.ui)) {
        return json;
    }
    json.children[0].props.ui = json.children[0].props.ui.mapKeys(key => {
        if (key.indexOf('reducer') >= 0) {
            return key;
        }
        return 'Component';
    });
    json.children[0].children = json.children[0].children.map(replaceUiKeyRecursively);
    return json;
};

export const snapshot = renderedComponent => {
    const componentJson = toJson(renderedComponent);
    const componentWithoutReduxUIKeys = removeReduxUIGeneratedParts(componentJson);
    return componentWithoutReduxUIKeys;
};

export const reset = component => () => {
    if (Boolean(component)) {
        component.unmount();
    }
    document.body.innerHTML = '';
};
