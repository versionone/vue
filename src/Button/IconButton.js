import React, {PropTypes} from 'react';
import * as CustomPropTypes from './../utilities/CustomPropTypes';
import Radium from './../utilities/Radium';
import SvgIcon from './../SvgIcon';
import transparent from './../utilities/Transparent';

const getStyles = props => ({
    root: {
        backgroundColor: props.backgroundColor,
        borderRadius: props.circle ? '50%' : '0px',
        cursor: 'pointer',
        lineHeight: 0.6,
    },
});

const handleIconClick = handler => evt => handler(evt);

const IconButton = (props) => {
    const {
        backgroundColor,
        circle,
    } = props;
    const styles = getStyles({
        backgroundColor,
        circle,
    });

    return (
        <div
            style={styles.root}
            onClick={handleIconClick(props.onClick)}
        >
            {React.createElement(props.icon, {
                color: props.color,
                hoverColor: props.hoverColor,
                width: props.width,
            })}
        </div>
    );
};
IconButton.defaultProps = {
    backgroundColor: transparent,
    circle: false,
    onClick: () => {
    },
    width: 24,
};
IconButton.propTypes = {
    backgroundColor: PropTypes.string,
    circle: PropTypes.bool,
    color: PropTypes.string,
    hoverColor: PropTypes.string,
    icon: CustomPropTypes.componentType(SvgIcon).isRequired,
    onClick: PropTypes.func,
    width: PropTypes.number,
};
export default Radium(IconButton);
