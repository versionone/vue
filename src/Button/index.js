import Button from './Button';

export default Button;
export {default as IconButton} from './IconButton';
