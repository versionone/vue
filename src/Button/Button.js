import React, {PropTypes} from 'react';
import {setOpacity, darken, toRgbaString} from '@andrew-codes/color-functions';
import Radium from './../utilities/Radium';
import ThemeProvider from './../Theme';
import transparent from './../utilities/Transparent';
import {create} from './../styles/Transitions';
import {getForegroundForBackground} from './../utilities/colorManipulator';
import * as ButtonSizes from './Sizes';
import * as ButtonTypes from './Types';

const darkenInvert = (foreground, background) => {
    const inverseBasicColorMultiplier = 0.35;
    const darkenedColor = toRgbaString(darken(background, inverseBasicColorMultiplier));
    return {
        ':hover': {
            background: darkenedColor,
            border: `1px solid ${darkenedColor}`,
            color: foreground,
        },
        background,
        border: `1px solid ${background}`,
        color: foreground,
    };
};
const getStylesBasedOnType = (props, context) => {
    const {
        altColor,
        basicColor,
        importantColor,
        darkInverseColor,
        lightInverseColor,
        normalBackground,
        textPrimaryColor,
    } = context.theme;
    const {
        disable,
        type,
    } = props;
    const inverseColors = [
        darkInverseColor,
        lightInverseColor,
    ];

    if (disable) {
        const disableColorOpacity = 0.3;
        const color = toRgbaString(setOpacity(textPrimaryColor, disableColorOpacity));
        return {
            ':hover': {
                background: normalBackground,
                color,
            },
            background: normalBackground,
            border: `1px solid ${transparent}`,
            color,
        };
    }
    if (type === ButtonTypes.basic) {
        return darkenInvert(darkInverseColor, basicColor);
    }
    if (type === ButtonTypes.important) {
        return darkenInvert(darkInverseColor, importantColor);
    }
    if (type === ButtonTypes.alt) {
        return darkenInvert(darkInverseColor, altColor);
    }
    if (type === ButtonTypes.basicAlt) {
        return {
            ':hover': {
                background: basicColor,
                border: `1px solid ${basicColor}`,
                color: darkInverseColor,
            },
            background: normalBackground,
            border: `1px solid ${textPrimaryColor}`,
            color: textPrimaryColor,
        };
    }

    if (type === ButtonTypes.special) {
        return {
            ':hover': {
                background: basicColor,
                border: `1px solid ${basicColor}`,
                color: darkInverseColor,
            },
            background: textPrimaryColor,
            border: `1px solid ${textPrimaryColor}`,
            color: normalBackground,
        };
    }

    const inverseBackground = getForegroundForBackground(normalBackground, inverseColors);
    const inverseForeground = getForegroundForBackground(inverseBackground, inverseColors);
    return {
        ':hover': {
            background: inverseBackground,
            color: inverseForeground,
        },
        background: normalBackground,
        border: `1px solid ${textPrimaryColor}`,
        color: textPrimaryColor,
    };
};
const getStyles = (props, context) => {
    const {
        normalRadius,
        basicFontFamily,
        bold,
        largeLineHeight,
        smallFontSize,
    } = context.theme;
    const {
        size,
    } = props;

    const fontSize = smallFontSize * size;
    const height = Math.ceil(fontSize * largeLineHeight);
    const borderRadiusMultiplier = 2;
    const borderRadius = normalRadius * borderRadiusMultiplier;
    const typeStyles = getStylesBasedOnType(props, context);

    return {
        root: {
            ':focus': {
                outline: 'none',
            },
            alignItems: 'flex-start',
            border: `1px solid ${transparent}`,
            borderRadius: `${borderRadius}px`,
            boxSizing: 'border-box',
            cursor: 'pointer',
            display: 'inline-block',
            fontFamily: basicFontFamily,
            fontSize: `${fontSize}px`,
            fontWeight: bold,
            height: `${height}px`,
            letterSpacing: '0.03em',
            lineHeight: `${largeLineHeight}px`,
            margin: '0',
            padding: '0 1em',
            textAlign: 'center',
            textShadow: 'none',
            transition: create('0.5s'),
            whiteSpace: 'no-wrap',
            ...typeStyles,
        },
    };
};
const handleClick = (handler, disable) => (evt) => {
    if (disable) {
        return;
    }
    handler(evt);
};

const Button = (props, context) => {
    const {
        disable,
        onClick,
        size,
        text,
        type,
    } = props;
    const styles = getStyles({
        disable,
        size,
        type,
    }, context);
    return (
        <button
            style={styles.root}
            onClick={handleClick(onClick, disable)}
        >
            {text}
        </button>
    );
};
Button.propTypes = {
    /**
     * Disables the button from responding to event handlers
     */
    disable: PropTypes.bool,
    /**
     * Click event handler; fired once a button is clicked
     */
    onClick: PropTypes.func,
    /**
     * Numeric value used as a multiplier to the button's size; 0.75, 1, and 1.5 as examples
     */
    size: PropTypes.number,
    /**
     * Text string displayed within Button
     */
    text: PropTypes.string,
    /**
     * Type of button
     */
    type: PropTypes.oneOf([
        ButtonTypes.standard,
        ButtonTypes.basic,
        ButtonTypes.important,
        ButtonTypes.alt,
        ButtonTypes.basicAlt,
        ButtonTypes.special,
    ]),
};
Button.defaultProps = {
    disable: false,
    onClick: () => {
    },
    size: ButtonSizes.normal,
    text: '',
    type: ButtonTypes.standard,
};
Button.contextTypes = {
    theme: PropTypes.shape(ThemeProvider.themeDefinition).isRequired,
};

export default Radium(Button);

