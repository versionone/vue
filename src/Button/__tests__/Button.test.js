import Button from '../Button';
import {getShallow, snapshot} from './../../../specHelpers/rendering';
import * as ButtonTypes from '../Types';

const shallowRenderButton = getShallow(Button);
const clickEvent = {
    test: true,
};

test('button is click-able with an event handler', () => {
    const onClick = jest.fn();
    const button = shallowRenderButton({
        onClick,
    });
    simulateClick(button, clickEvent);
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith(clickEvent);
});

test('standard button', () => {
    const buttonWithDefaultType = shallowRenderButton({
        text: 'Click me',
    });
    expect(snapshot(buttonWithDefaultType)).toMatchSnapshot();

    const button = shallowRenderButton({
        text: 'Click me',
        type: ButtonTypes.standard,
    });
    expect(snapshot(button)).toMatchSnapshot();
});

test('basic button', () => {
    const button = shallowRenderButton({
        text: 'Click me',
        type: ButtonTypes.basic,
    });
    expect(snapshot(button)).toMatchSnapshot();
});

test('important button', () => {
    const button = shallowRenderButton({
        text: 'Click me',
        type: ButtonTypes.important,
    });
    expect(snapshot(button)).toMatchSnapshot();
});

test('alternative button', () => {
    const button = shallowRenderButton({
        text: 'Click me',
        type: ButtonTypes.alt,
    });
    expect(snapshot(button)).toMatchSnapshot();
});

test('basic alternative button', () => {
    const button = shallowRenderButton({
        text: 'Click me',
        type: ButtonTypes.basicAlt,
    });
    expect(snapshot(button)).toMatchSnapshot();
});

test('special button', () => {
    const button = shallowRenderButton({
        text: 'Click me',
        type: ButtonTypes.special,
    });
    expect(snapshot(button)).toMatchSnapshot();
});

test('button can be resized', () => {
    const button = shallowRenderButton({
        size: 0.75,
    });
    expect(snapshot(button)).toMatchSnapshot();
});

test('button can be disabled', () => {
    const onClick = jest.fn();
    const button = shallowRenderButton({
        disabled: true,
        onClick,
    });
    simulateClick(button);
    expect(onClick).not.toBeCalled();
    expect(snapshot(button)).toMatchSnapshot();
});

function simulateClick(wrapper, evt = {}) {
    wrapper.simulate('click', evt);
}
