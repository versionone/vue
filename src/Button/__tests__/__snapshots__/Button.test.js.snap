// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alternative button 1`] = `
<button
  data-radium={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "background": "#eaab00",
      "border": "1px solid #eaab00",
      "borderRadius": "12px",
      "boxSizing": "border-box",
      "color": "#ffffff",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "'Proxima Nova', 'Lucida Sans Unicode', 'Lucida Grande', sans-serif",
      "fontSize": "14px",
      "fontWeight": undefined,
      "height": "32px",
      "letterSpacing": "0.03em",
      "lineHeight": "2.285",
      "margin": "0",
      "padding": "0 1em",
      "textAlign": "center",
      "textShadow": "none",
      "transition": "all 0.5s linear 0ms",
      "whiteSpace": "no-wrap",
    }
  }
>
  Click me
</button>
`;

exports[`basic alternative button 1`] = `
<button
  data-radium={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "background": "#ffffff",
      "border": "1px solid #31363e",
      "borderRadius": "12px",
      "boxSizing": "border-box",
      "color": "#31363e",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "'Proxima Nova', 'Lucida Sans Unicode', 'Lucida Grande', sans-serif",
      "fontSize": "14px",
      "fontWeight": undefined,
      "height": "32px",
      "letterSpacing": "0.03em",
      "lineHeight": "2.285",
      "margin": "0",
      "padding": "0 1em",
      "textAlign": "center",
      "textShadow": "none",
      "transition": "all 0.5s linear 0ms",
      "whiteSpace": "no-wrap",
    }
  }
>
  Click me
</button>
`;

exports[`basic button 1`] = `
<button
  data-radium={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "background": "#00a9e0",
      "border": "1px solid #00a9e0",
      "borderRadius": "12px",
      "boxSizing": "border-box",
      "color": "#ffffff",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "'Proxima Nova', 'Lucida Sans Unicode', 'Lucida Grande', sans-serif",
      "fontSize": "14px",
      "fontWeight": undefined,
      "height": "32px",
      "letterSpacing": "0.03em",
      "lineHeight": "2.285",
      "margin": "0",
      "padding": "0 1em",
      "textAlign": "center",
      "textShadow": "none",
      "transition": "all 0.5s linear 0ms",
      "whiteSpace": "no-wrap",
    }
  }
>
  Click me
</button>
`;

exports[`button can be disabled 1`] = `
<button
  data-radium={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "background": "#ffffff",
      "border": "1px solid rgba(0,0,0,0)",
      "borderRadius": "12px",
      "boxSizing": "border-box",
      "color": "rgba(49,54,62,0.3)",
      "cursor": "not-allowed",
      "display": "inline-block",
      "fontFamily": "'Proxima Nova', 'Lucida Sans Unicode', 'Lucida Grande', sans-serif",
      "fontSize": "14px",
      "fontWeight": undefined,
      "height": "32px",
      "letterSpacing": "0.03em",
      "lineHeight": "2.285",
      "margin": "0",
      "padding": "0 1em",
      "textAlign": "center",
      "textShadow": "none",
      "transition": "all 0.5s linear 0ms",
      "whiteSpace": "no-wrap",
    }
  }
/>
`;

exports[`button can be resized 1`] = `
<button
  data-radium={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "background": "#ffffff",
      "border": "1px solid #31363e",
      "borderRadius": "12px",
      "boxSizing": "border-box",
      "color": "#31363e",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "'Proxima Nova', 'Lucida Sans Unicode', 'Lucida Grande', sans-serif",
      "fontSize": "10.5px",
      "fontWeight": undefined,
      "height": "24px",
      "letterSpacing": "0.03em",
      "lineHeight": "2.285",
      "margin": "0",
      "padding": "0 1em",
      "textAlign": "center",
      "textShadow": "none",
      "transition": "all 0.5s linear 0ms",
      "whiteSpace": "no-wrap",
    }
  }
/>
`;

exports[`important button 1`] = `
<button
  data-radium={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "background": "#ea6c02",
      "border": "1px solid #ea6c02",
      "borderRadius": "12px",
      "boxSizing": "border-box",
      "color": "#ffffff",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "'Proxima Nova', 'Lucida Sans Unicode', 'Lucida Grande', sans-serif",
      "fontSize": "14px",
      "fontWeight": undefined,
      "height": "32px",
      "letterSpacing": "0.03em",
      "lineHeight": "2.285",
      "margin": "0",
      "padding": "0 1em",
      "textAlign": "center",
      "textShadow": "none",
      "transition": "all 0.5s linear 0ms",
      "whiteSpace": "no-wrap",
    }
  }
>
  Click me
</button>
`;

exports[`special button 1`] = `
<button
  data-radium={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "background": "#31363e",
      "border": "1px solid #31363e",
      "borderRadius": "12px",
      "boxSizing": "border-box",
      "color": "#ffffff",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "'Proxima Nova', 'Lucida Sans Unicode', 'Lucida Grande', sans-serif",
      "fontSize": "14px",
      "fontWeight": undefined,
      "height": "32px",
      "letterSpacing": "0.03em",
      "lineHeight": "2.285",
      "margin": "0",
      "padding": "0 1em",
      "textAlign": "center",
      "textShadow": "none",
      "transition": "all 0.5s linear 0ms",
      "whiteSpace": "no-wrap",
    }
  }
>
  Click me
</button>
`;

exports[`standard button 1`] = `
<button
  data-radium={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "background": "#ffffff",
      "border": "1px solid #31363e",
      "borderRadius": "12px",
      "boxSizing": "border-box",
      "color": "#31363e",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "'Proxima Nova', 'Lucida Sans Unicode', 'Lucida Grande', sans-serif",
      "fontSize": "14px",
      "fontWeight": undefined,
      "height": "32px",
      "letterSpacing": "0.03em",
      "lineHeight": "2.285",
      "margin": "0",
      "padding": "0 1em",
      "textAlign": "center",
      "textShadow": "none",
      "transition": "all 0.5s linear 0ms",
      "whiteSpace": "no-wrap",
    }
  }
>
  Click me
</button>
`;

exports[`standard button 2`] = `
<button
  data-radium={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "background": "#ffffff",
      "border": "1px solid #31363e",
      "borderRadius": "12px",
      "boxSizing": "border-box",
      "color": "#31363e",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "'Proxima Nova', 'Lucida Sans Unicode', 'Lucida Grande', sans-serif",
      "fontSize": "14px",
      "fontWeight": undefined,
      "height": "32px",
      "letterSpacing": "0.03em",
      "lineHeight": "2.285",
      "margin": "0",
      "padding": "0 1em",
      "textAlign": "center",
      "textShadow": "none",
      "transition": "all 0.5s linear 0ms",
      "whiteSpace": "no-wrap",
    }
  }
>
  Click me
</button>
`;
